### ISSUE TITLE
TypeError: unorderable types: VNodeBase() < VNodeBase()

### ISSUE THREAD

**Issue by SegundoBob** [leo-editor/leo-editor#224]
Outline of the Test Scenario
-----------------------------------
Create a new empty Leo-Editor file.
Change its single node to an @file node for a new file.  Say x.txt.
Put "@others" in the body of the @file node.
Create one child node of the @file node.
Save the new Leo-Editor file.
This should create x.txt.
With the @file node collapsed and the focus on the @file node, import (Alt-F, I) x.txt.
This should make the child of both @file nodes clones of one another.

Now you are stuck.

You can't eliminate the unwanted clone by copying it to a non-clone and then deleting the clone.  That is, copy (Shift-Ctrl-C), paste (Shift-Ctrl-V), then delete one clone.  Because, the paste encounters this bug.

You can't save your Leo-Editor file because save encounters this bug.

If you try saving on exit, your Leo-Editor file is left marked open becuase this bug borks the save and the file close.

These problems leave you with the Leo-Editor file in the state it was when you saved it most recently before doing the file import.  You could also delete the imported subtree and then successfully save your Leo-Editor file.

How I Encountered the Problem
----------------------------------------
I started modifying a Python script.  Then I realized that my changes were extensive and I wanted to keep the working version around.  So I changed the name of the Python script.  Then I imported the old script.

Trace
-------
```
Traceback (most recent call last):
  File "/pri/git/leo-editor/leo/core/leoCommands.py", line 5088, in doCommand
    val = command(event)
  File "/pri/git/leo-editor/leo/core/leoGlobals.py", line 1335, in wrapper
    func(self, event=event)
  File "/pri/git/leo-editor/leo/core/leoCommands.py", line 933, in save
    c.fileCommands.save(c.mFileName)
  File "/pri/git/leo-editor/leo/core/leoFileCommands.py", line 1367, in save
    ok = self.write_Leo_file(fileName, False) # outlineOnlyFlag
  File "/pri/git/leo-editor/leo/core/leoFileCommands.py", line 1784, in write_Leo_file
    structure_errors = c.checkOutline()
  File "/pri/git/leo-editor/leo/core/leoCommands.py", line 3167, in checkOutline
    structure_errors = c.checkGnxs()
  File "/pri/git/leo-editor/leo/core/leoCommands.py", line 3214, in checkGnxs
    aList = sorted(d.get(gnx))
TypeError: unorderable types: VNodeBase() < VNodeBase()
```
Test System
---------------
Xubuntu64 14.04
```
Leo Log Window
Leo 5.1-final, build 20151122092056, Sun Nov 22 09:20:56 CST 2015
Git repo info: branch = None, commit = 918b7b9e1ecd
Python 3.4.3, PyQt version 5.2.1
linux
```
and
```
Leo Log Window
Leo 5.1-final, build 20151130174247, Mon Nov 30 17:42:47 CST 2015
Git repo info: branch = None, commit = 71b1caffcfaa
Python 3.4.3, PyQt version 5.2.1
linux
```

**Comment by edreamleo** [comment]
​​On Tue, Dec 1, 2015 at 4:13 PM, SegundoBob <notifications@github.com>
wrote:

> Tracebac
> ​​
> k (most recent call last):
> ​[snip]​
>   File "/pri/git/leo-editor/leo/core/leoCommands.py", line 3214, in checkGnxs
>     aList = sorted(d.get(gnx))
> TypeError: unorderable types: VNodeBase() < VNodeBase()
>
> ​Good catch. ​
The fix is likely to be aList = list(d.get(gnx)), which does not require
that
​vnodes be orderable.​
​If you would, please verify that the fix works for you.  I shall then push
the correction.  ​

Thanks.

Edward

**Comment by SegundoBob** [comment]
Your suggested change eliminates the error that caused the stack trace.  But now there is a probably incorrect complaint that there are two GNX errors:
```
Leo Log Window
Leo 5.1-final, build 20151130174247, Mon Nov 30 17:42:47 CST 2015
Git repo info: branch = None, commit = 71b1caffcfaa
Python 3.4.3, PyQt version 5.2.1
linux
setting leoID from os.getenv('USER'): 'bob'
load dir: /pri/git/leo-editor/leo/core
global config dir: /pri/git/leo-editor/leo/config
home dir: /home/bob
reading settings in /pri/git/leo-editor/leo/config/leoSettings.leo
reading settings in /home/bob/.leo/myLeoSettings.leo
Found /help
reading settings in /sec/tmp/e-ck-gnxs.leo
reading: /sec/tmp/e-ck-gnxs.leo
read 1 files in 0.02 seconds
read outline in 0.04 seconds

reading: @file /sec/tmp/e-ck-gnxs-1.txt
multiple vnodes with gnx: 'bob.20151201132538.1'
new gnx: bob.20151201132538.1 <VNode 7f4c416192b0 @file e-ck-gnxs-1.txt>
new gnx: bob.20151201132538.1 <VNode 7f4c40bd6c50 @file /sec/tmp/e-ck-gnxs-1.txt>
check-outline ERROR! e-ck-gnxs.leo 4 nodes, 2 gnx errors, 0 structure errors
wrote: e-ck-gnxs-1.txt
wrote: e-ck-gnxs-1.txt
saved: e-ck-gnxs.leo
```

**Comment by SegundoBob** [comment]
After looking at the files, I think the error message is OK.  There really is a GNX problem, though it seems harmless.  The problem is that two files (whether they are actually the same file or separate files isn't important) contain the same GNX.

Despite the error message, everything works as expected and the problem is easily eliminated by copying one clone and then deleting the clone.

**Comment by edreamleo** [comment]
On Wed, Dec 2, 2015 at 12:00 PM, SegundoBob <notifications@github.com>
wrote:

> After looking at the files, I think the error message is OK. There really
> is a GNX problem, though it seems harmless. The problem is that two files
> (whether they are actually the same file or separate files isn't important)
> contain the same GNX.
>
> Despite the error message, everything works as expected and the problem is
> easily eliminated by copying one clone and then deleting the clone.
>
​Thanks for the confirmation.  I'll up the proposed change next.

EKR

**Comment by edreamleo** [comment]
Fixed at d1169b93.
